src/gpg/eopen.py

```diff
@@ -0,0 +1,72 @@
+import gpg
+import os
+import sys
+import time
+import textwrap
+
+DEBUG= True if (os.environ['DEBUG']=='yes') else False
+
+def print_signatures(verify_result):
+    c=gpg.Context()
+    for signature in verify_result.signatures:
+        user = c.get_key(signature.fpr).uids[0].uid
+        fpr = signature.fpr
+        signed_time = time.ctime(signature.timestamp)
+
+        message = '''
+                    Good signature from {user}
+                    with key {fingerprint}
+                    made at {time}
+                    '''.format(user=user,
+                                fingerprint=fpr,
+                                time=signed_time)
+
+        print(textwrap.dedent(message))
+
+def eopen(sealed_file_path, output_file_path):
+    """
+    sign and encrypt file_path to the recipents
+    """
+
+    try:
+        c = gpg.Context()
+        with open(sealed_file_path, "rb") as cfile:
+            try:
+                plaintext, result, verify_result = c.decrypt(cfile,verify=True)
+                
+                if(DEBUG):
+                    print(verify_result, result)
+                
+                print_signatures(verify_result)
+
+            except gpg.errors.BadSignatures:
+                print("Could not verify signatures")
+                exit(3)
+                
+        while(os.path.exists(output_file_path)):
+            print("File {output} already exists".format(output=output_file_path))
+            print("Do you want to overwrite?(y/N)", end=">")
+            
+            ans=input()
+            if(ans.lower()=='y'):
+                break
+            else:
+                print("New file name",end=":")
+                output_file_path=input()
+            
+        with open(output_file_path, "wb") as nfile:
+            nfile.write(plaintext)
+
+        # verify signatures
+
+    except BaseException:
+        if(DEBUG):
+            raise
```

Comment:
Try to finish this task as soon as possible: https://github.com/EasyGnuPG/pgpg/issues/54
because it may also help you with development.

Reply:
opened #62 for the same.